build.go

```diff
@@ -30,10 +30,20 @@ func Build(
 ) packit.BuildFunc {
 	return func(context packit.BuildContext) (packit.BuildResult, error) {
 		logger.Title("%s %s", context.BuildpackInfo.Name, context.BuildpackInfo.Version)
+		var projectPath string
+		var err error
 
-		projectPath, err := buildpackYMLParser.ParseProjectPath(filepath.Join(context.WorkingDir, "buildpack.yml"))
-		if err != nil {
-			return packit.BuildResult{}, err
+		projectPath = os.Getenv("BP_DOTNET_PROJECT_PATH")
+		if projectPath == "" {
+			projectPath, err = buildpackYMLParser.ParseProjectPath(filepath.Join(context.WorkingDir, "buildpack.yml"))
+			if err != nil {
+				return packit.BuildResult{}, err
+			}
+
+			if projectPath != "" {
+				logger.Subprocess("WARNING: Setting the project path through buildpack.yml will be deprecated soon in Dotnet Publish Buildpack v1.0.0")
```

Comment:
I believe in our deprecation warnings we want to do [something like this](https://github.com/paketo-buildpacks/mri/blob/d09a1e154eb5925d0cc84cc4cdb5fc2b85a383e5/build.go#L69-L70), to avoid hardcoding the buildpack version.